tests/spec/SpecHelper.js

```diff
@@ -56,6 +56,34 @@ export function makeRSDoc(opts, src, style = "") {
     iframes.push(ifr);
   });
 }
+/**
+ * Used to get errors and warnings from a spec.
+ */
+class UIMessageFilters extends Map {
+  /**
+   * @param {"warnings" | "errors"} type
+   */
+  constructor(type) {
+    super();
+    this.type = type;
+  }
+  /**
+   * @param {string} pluginName
+   * @returns (Document) => Array<RespecError>
+   */
+  filter(pluginName) {
+    if (this.has(pluginName)) {
+      return this.get(pluginName);
+    }
```

Comment:
Not following why we might want a Map here.

Reply:
You always need a map :) ... but I'm thinking I'll just make it internal, not extension.